contracts/src/locker.cairo

```diff
@@ -0,0 +1,123 @@
+use starknet::ContractAddress;
+
+use openzeppelin::token::erc20::interface::{ERC20ABIDispatcher, ERC20ABIDispatcherTrait};
+
+#[starknet::interface]
+trait ILocker<TContractState> {
+    fn lock_liquidity(ref self: TContractState, lp_address: ContractAddress, amount: u256);
+    fn lockLiquidity(ref self: TContractState, lpAddress: ContractAddress, amount: u256);
+    fn unlock_liquidity(ref self: TContractState, lp_address: ContractAddress, lock_time: u64);
+    fn get_locked_amount(self: @TContractState, lp_address: ContractAddress, owner: ContractAddress, lock_time: u64) -> u256;
+    fn get_time_left(self: @TContractState, lp_address: ContractAddress, owner: ContractAddress, lock_time: u64) -> u64;
+}
+
+#[starknet::contract]
+mod Locker {
+    use starknet::{ContractAddress, get_contract_address, get_caller_address, get_block_timestamp};
+    use super::{ERC20ABIDispatcher, ERC20ABIDispatcherTrait};
+
+    #[storage]
+    struct Storage {
+        lock_time: u64, // Locking period. 6 months in timestamp
+        locks: LegacyMap<(ContractAddress, ContractAddress, u64), u256>, // LP -> Owner -> Lock time -> amount
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState, lock_time: u64) {
+        self.lock_time.write(lock_time);
+    }
+
+    #[external(v0)]
+    impl Locker of super::ILocker<ContractState> {
+        // Locks liquidity in this smart contract.
+        fn lock_liquidity(ref self: ContractState, lp_address: ContractAddress, amount: u256) {
+            assert(amount > 0, 'Zero lock');
```

Comment:
amount is of type u256 and can't be negative, prefer an equality check instead

also: can you create an `Errors` module, a `const ZERO_LOCK:felt252 = 'Zero lock`  and use this instead? Same for all the future asserts

```suggestion
            assert(amount != 0, errors::ZERO_LOCK);
```